catalog/plugin/catalog-plugin-metacardingest-network/src/test/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeFactoryTest.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.metacard.util;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.when;
+import static ddf.catalog.data.AttributeType.AttributeFormat.BINARY;
+import static ddf.catalog.data.AttributeType.AttributeFormat.BOOLEAN;
+import static ddf.catalog.data.AttributeType.AttributeFormat.DATE;
+import static ddf.catalog.data.AttributeType.AttributeFormat.DOUBLE;
+import static ddf.catalog.data.AttributeType.AttributeFormat.FLOAT;
+import static ddf.catalog.data.AttributeType.AttributeFormat.GEOMETRY;
+import static ddf.catalog.data.AttributeType.AttributeFormat.INTEGER;
+import static ddf.catalog.data.AttributeType.AttributeFormat.LONG;
+import static ddf.catalog.data.AttributeType.AttributeFormat.OBJECT;
+import static ddf.catalog.data.AttributeType.AttributeFormat.SHORT;
+import static ddf.catalog.data.AttributeType.AttributeFormat.STRING;
+import static ddf.catalog.data.AttributeType.AttributeFormat.XML;
+
+import java.io.Serializable;
+import java.util.List;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType;
+
+/**
+ * Validate the behavior for {@link AttributeFactory}.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class AttributeFactoryTest {
+
+    private static final String SAMPLE_JSON_STRING = "{ \"id\": \"j8k767bjav592j2\"}";
+
+    private static final String LANGUAGE = "language";
+
+    private static final String COMMA_SEPARATED_LANGUAGES = "English, French, German";
```

Comment:
To make the test a little less brittle, we probably should build this list using the constants instead.